metrics/metrics-tests/src/main/java/org/apache/james/metrics/tests/RecordingMetric.java

```diff
@@ -52,13 +51,20 @@ public void add(int i) {
 
     @Override
     public void remove(int i) {
-        value.addAndGet(-1 * i);
+        value.updateAndGet(currentValue -> subtractFrom(currentValue, i));
```

Comment:
subsctractFrom is not thread safe anymore.

Given that this is used only in test, I would advocate making all public method synchronized and relying on `int` rather than `AtomicInteger`

Reply:
My bad, read too fast